### PR TITLE
feat(progress-indicator): update to use carbon icons

### LINKED ISSUE #5086: [Progress indicator]: Add `viewBox` attr to svg to allow scaling
### Summary

We are bumping up icon sizes to 20px by 20px for the expressive theme changes. Ticket here: https://github.com/carbon-design-system/ibm-dotcom-library/issues/1039

However the some svgs for the progress indicator component don't scale correctly.

Screen Shot 2020-01-16 at 10 47 26 AM

Looking at the code it seems that there are no attributes for the `svg` elements 
```
if (current) {
      return (
        <svg>
          <path d="M 7, 7 m -7, 0 a 7,7 0 1,0 14,0 a 7,7 0 1,0 -14,0" />
          <title>{description}</title>
        </svg>
      );
    }
    
     ....

    return (
      <svg>
        <title>{description}</title>
        <path d="M8 1C4.1 1 1 4.1 1 8s3.1 7 7 7 7-3.1 7-7-3.1-7-7-7zm0 13c-3.3 0-6-2.7-6-6s2.7-6 6-6 6 2.7 6 6-2.7 6-6 6z" />
      </svg>
```

If the `viewBox` attribute is added to the `svg` elements, that would allow for the `current` and `incomplete` indicator icons to scale to 20x20.

### Justification

This is needed for the expressive theme changes by the DDS team. 

### "Must have" functionality

- svgs should have `viewBox` attribute defined

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

It would be great to have this in during our next sprint (Sprint 20-02) which ends on January 31st.

_NB: The Carbon team will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_

### PR DESCRIPTION
**Pull request by annawen1**
Closes #5086

Updates the `<ProgressIndicator />` to pull icons from the carbon icons package instead of hardcoding the svg paths. Used the `<RadioButton16 />` and `<RadioButtonChecked16 />` icons - I didn't notice any filled circle icons so I had to style the `<RadioButtonChecked16 />` stroke. 

Screen Shot 2020-01-21 at 1 33 44 PM

#### Changelog

**New**

- import `<RadioButtonChecked16 />` for current progress step icon and `<RadioButton16 />` for complete and disabled progress step icons

**Changed**

- removed the 14px width and height styling for the current icon svg, doesn't match the other icons
- adjust the `<RadioButtonChecked16 />` to produce the filled circle we need for complete-progress-step icon

#### Testing / Reviewing

Testing should ensure `<ProgressIndicator />`  is consistent with designs

### CODE REVIEW

**Review by tw15egan — APPROVED** [review]
LGTM 👍

Wondering if we should update the vanilla `.hbs` files with the correct SVG assets as well.

File: packages/components/src/components/progress-indicator/_progress-indicator.scss [hunk 1]
```diff
@@ -142,10 +142,13 @@
   }
 
   .#{$prefix}--progress-step--current svg {
-    width: 14px;
-    height: 14px;
+    stroke: $interactive-01;
     fill: $interactive-01;
     margin-top: rem(9.5px);
+
+    path:last-of-type {
```
Comment (joshblack) [inline packages/components/src/components/progress-indicator/_progress-indicator.scss:149]: Just curious, what is this selector being used for? 👀

Comment (tw15egan) [reply packages/components/src/components/progress-indicator/_progress-indicator.scss:149]: Looks like it is being used to fill in the circle by increasing the width of the checked circle

Comment (annawen1) [reply packages/components/src/components/progress-indicator/_progress-indicator.scss:149]: yeah...I couldn't find a Filled Circle Icon in the carbon/icon library...so I'm improvised 😅

should I try to get a filled circle icon in the icon package instead?

Comment (asudoh) [reply packages/components/src/components/progress-indicator/_progress-indicator.scss:149]: > should I try to get a filled circle icon in the icon package instead?

👍

Comment (joshblack) [reply packages/components/src/components/progress-indicator/_progress-indicator.scss:149]: cc @laurenmrice do you know of an icon we could use or will this work as is? 👀

Comment (laurenmrice) [reply packages/components/src/components/progress-indicator/_progress-indicator.scss:149]: Yeah we do not have a filled circle icon. I think we can do this for now and I can talk with conrad about possibly adding one to our library for the future. @joshblack

**Comment by abbeyhrt** [comment]
@annawen1 thank you for doing this! I might be wrong but it seems like the filled blue circe looks a tiny bit bigger than the one we have right now

**Comment by tw15egan** [comment]
@abbeyhrt I think that's because the current one is an incorrect `14px`, and the rest are `16px`.

**Comment by laurenmrice** [comment]
Wait a minute we do have a `dot mark` icon!! We could use this for the `current state` https://carbon-elements.netlify.com/icons/examples/preview/#32%2Fdot-mark%20(Downsized%20to%2016)

**Comment by abbeyhrt** [comment]
@tw15egan makes sense!

**Comment by annawen1** [comment]
> https://carbon-elements.netlify.com/icons/examples/preview/#32%2Fdot-mark%20(Downsized%20to%2016)

@laurenmrice the dotmark icon doesn't take up the entire svg, there's some white space around it. Is there anyway to change this?

**Comment by tw15egan** [comment]
@laurenmrice @annawen1 I also tried using that icon when I tested this PR, but yeah it is sized smaller than the other 16px icons (makes sense, since it is a dot mark). We would need a full 16px circle

**Comment by laurenmrice** [comment]
Ahh okay I see. Alright I will still talk to Conrad about this to see the reason behind the dot mark size and if he can adjust that one to match the other circular icons, or if we need to make a separate icon.

**Comment by laurenmrice** [comment]
Update on filled icon for current state: Conrad is going to make a `circle--filled` icon and it will be in the repo in the next couple of weeks, so we can update it then in a separate pr.

**Comment by annawen1** [comment]
@laurenmrice awesome!

**Comment by asudoh** [comment]
@laurenmrice Does this mean this PR itself is good to go...? Thanks!

**Review by abbeyhrt — APPROVED** [review]

**Comment by tw15egan** [comment]
@laurenmrice is this good to merge?

**Review by laurenmrice — CHANGES_REQUESTED** [review]
Can you double check that the filled icon is 16 like the rest of them, i know it currently is not an actual icon but just making sure its still the right size. It does still seem to be a little larger. 

Screen Shot 2020-01-27 at 11 53 32 AM

The rest of the icons look good

**Comment by annawen1** [comment]
@laurenmrice 

I removed the margin-top that was being set on the completed icon, looks closer in size now
Screen Shot 2020-01-28 at 9 29 54 AM

**Review by laurenmrice — APPROVED** [review]
lgtm thank you ! 🙌🏻

**Pull request merged**